### PR TITLE
Clean up memory allocated when reading messages in H5Dlayout on error

### PR DESCRIPTION
**Pull request by glennsong09**

### CODE REVIEW

File: release_docs/RELEASE.txt [hunk 1]
```diff
@@ -123,6 +123,16 @@ New Features
...
+    - Fixed memory leaks that could occur when reading a dataset from a
+      malformed file
+
+      When attempting to read layout, pline, and efl information for a
+      dataset, memory leaks could occur if attempting to read pline/efl
+      information threw an error, which is due to the memory that was 
+      allocated for pline and efl not being properly cleaned up on error.  
+	
+	Fixes Github issue #2602
+
```
Comment (bmribler) [inline release_docs/RELEASE.txt:135]: Should this be lined up with the rest?

File: src/H5Dlayout.c [hunk 2]
```diff
@@ -661,10 +665,17 @@ H5D__layout_oh_read(H5D_t *dataset, hid_t dapl_id, H5P_genplist_t *plist)
...
+    if (ret_value < 0) {
+        if (pline_copied)
+            if (H5O_msg_reset(H5O_PLINE_ID, &dataset->shared->dcpl_cache.pline) < 0)
+                HDONE_ERROR(H5E_DATASET, H5E_CANTRESET, FAIL, "unable to reset pipeline info")
+        if (layout_copied)
+            if (H5O_msg_reset(H5O_LAYOUT_ID, &dataset->shared->layout) < 0)
+                HDONE_ERROR(H5E_DATASET, H5E_CANTRESET, FAIL, "unable to reset layout info")
+        if (efl_copied)
+            if (H5O_msg_reset(H5O_EFL_ID, &dataset->shared->dcpl_cache.efl) < 0)
+                HDONE_ERROR(H5E_DATASET, H5E_CANTRESET, FAIL, "unable to reset efl message")
```
Comment (bmribler) [inline src/H5Dlayout.c:677]: "unable to reset efl info"?  Or the other two "info" should be "message"?

File: src/H5Dlayout.c [hunk 3]
```diff
@@ -630,6 +633,7 @@ H5D__layout_oh_read(H5D_t *dataset, hid_t dapl_id, H5P_genplist_t *plist)
         /* Retrieve the EFL  message */
```
Comment (bmribler) [inline src/H5Dlayout.c]: Is there an extra ' ' (space) here?  It looks so.

**Review by bmribler — APPROVED** [review]
All my comments are very minor, so I approve it, but if you agree, please make the changes.

File: release_docs/RELEASE.txt (hunk 1 above)
Comment (glennsong09) [reply release_docs/RELEASE.txt:135]: I'm not sure how that happened, I can line it up with the rest.

File: src/H5Dlayout.c (hunk 3 above)
Comment (glennsong09) [reply src/H5Dlayout.c]: I'm not quite sure where the extra space is. Could you point me to it? Thank you!

Comment (bmribler) [reply src/H5Dlayout.c]: If you don't see it between EFL and message then it's just my illusion. :D

Comment (glennsong09) [reply src/H5Dlayout.c]: Oh, now I see what you are referring to.

File: src/H5Dlayout.c (hunk 2 above)
Comment (glennsong09) [reply src/H5Dlayout.c:677]: I think in order to keep it the same as in develop, it wouldn't make sense to change it right now.

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**